extras/kms/kms.go

```diff
@@ -289,55 +289,169 @@ func (k *Kms) CreateKeys(ctx context.Context, scopeId string, purposes []KeyPurp
 		opts.withRandomReader = rand.Reader
 	}
 
+	r, w, localTx, err := k.txFromOpts(ctx, opt...)
+	if err != nil {
+		return fmt.Errorf("%s: %w", op, err)
+	}
+
+	if _, err := createKeysTx(ctx, r, w, rootWrapper, opts.withRandomReader, scopeId, purposes...); err != nil {
+		if localTx != nil {
+			if rollBackErr := localTx.Rollback(ctx); rollBackErr != nil {
+				err = multierror.Append(err, rollBackErr)
+			}
+		}
+		return fmt.Errorf("%s: %w", op, err)
+	}
+	if localTx != nil {
+		if err := localTx.Commit(ctx); err != nil {
+			return fmt.Errorf("%s: unable to commit transaction: %w", op, err)
+		}
+	}
+
+	return nil
+}
+
+// RotateKeys will rotate the scope's root key version and all DEKs for the
+// current KeyPurpose(s) of the KMS.
+//
+// If an optional WithRewrap(...) is requested, then all keys will be re-encrypted
```

Comment:
Shouldn't we also have an ability to rewrap without requiring rotation? That is, a corresponding RewrapKeys function?

Reply:
it might be nice to have both and we can always add that in the future.   At this point, I wanted to get rotation implemented first.  perhaps I'll update the PR title to reflect that it's rotation with optional rewrap.